pymeasure/instruments/racal/racal1992.py

```diff
@@ -0,0 +1,327 @@
+            set_process=(lambda enable: {True: "E", False: "D"}[enable])
+        )
+
+    delay_time = Instrument.control(
+            " RDT",
+            " SDT %f",
+            """ Control delay time.""",
+            get_process=(lambda v: Racal1992.decode(v, "DT"))
+        )
+
+    special_function_enable = Instrument.setting(
+            " SF%s",
+            """ Enable or disable special function. True=enable, False=disable""",
+            set_process=(lambda enable: {True: "E", False: "D"}[enable])
+        )
+
+    # FIXME: not tested on real instrument!
+    special_function_number = Instrument.control(
+            " RSF",
+            " S%d",
+            """ Control special function.""",
+            get_process=(lambda v: Racal1992.decode(v, "SF"))
+        )
+
+    # FIXME: not tested on real instrument!
+    total_so_far = Instrument.measurement(
+            " RF",
+            """ Total number of events so far.""",
+            get_process=(lambda v: Racal1992.decode(v, "RF"))
+        )
+
+    software_version = Instrument.measurement(
+            " RMS",
+            "Instrument software version",
+            get_process=(lambda v: Racal1992.decode(v, "MS"))
+        )
+
+    gpib_software_version = Instrument.measurement(
+            " RGS",
+            "GPIB software version",
+            get_process=(lambda v: Racal1992.decode(v, "GS"))
+        )
+
+    device_type = Instrument.measurement(
+            " RUT",
+            """Unit device type. Should return 1992 for a Racal-Dana 1992
+            or 1991 for a Racal-Dana 1991.""",
+            get_process=(lambda v: Racal1992.decode(v, "UT"))
+        )
+
+    math_mode = Instrument.setting(
+            " M%s",
+            """ Enable or disable math mode. True=enable, False=disable""",
+            set_process=(lambda enable: {True: "E", False: "D"}[enable])
+        )
+
+    math_x = Instrument.control(
+            " RMX",
+            " SMX %f",
+            """ Control math constant X.""",
+            get_process=(lambda v: Racal1992.decode(v, "MX"))
+        )
+
+    math_z = Instrument.control(
+            " RMZ",
+            " SMZ %f",
+            """ Control math constant Z.""",
+            get_process=(lambda v: Racal1992.decode(v, "MZ"))
+        )
+
+    trigger_level_a = Instrument.control(
+            " RLA",
+            " SLA %f",
+            """ Control trigger level for channel A""",
+            get_process=(lambda v: Racal1992.decode(v, "LA"))
+        )
+
+    trigger_level_b = Instrument.control(
+            " RLB",
+            " SLB %f",
+            """ Control trigger level for channel B""",
+            get_process=(lambda v: Racal1992.decode(v, "LB"))
+        )
+
+    def __init__(self, adapter, name="Racal-Dana 1992", **kwargs):
+        kwargs.setdefault('write_termination', '\r\n')
+
+        super().__init__(
+            adapter,
+            name,
+            **kwargs
+        )
+
+    def read(self):
```

Comment:
@tomverbeure, I have read your blog about the weird problem that happened in your instrument, very strange indeed. Although I don't have a possible explanation, I would suggest to re-implement the `write` method such that you can add a space in front of any command to write. This would avoid to disseminate spaces in all the command definition and, if one day the explanation is found, you can easily change back as appropriate

Reply:
I've refactored all the leading spaces into a new `write` method.